Build fails: cannot find tbb.pc.in

When included into a superproject using `add_subdirectory`(for example through `FetchContent`), oneTBB 2021.3.0 fails to build with

```
CMake Error: File /home/xxx/myproject/integration/pkg-config/tbb.pc.in does not exist.
```

The offending line is the following:

https://github.com/oneapi-src/oneTBB/blob/ac3d43b691b69f943fee95668aa373c13d0d0d47/src/tbb/CMakeLists.txt#L154

The problem here is usage of `${CMAKE_SOURCE_DIR}`. This variable refers to the directory where top-level cmake file is located. `${PROJECT_SOURCE_DIR}` should be used instead, it always refers to the oneTBB directory.

--- a/src/tbb/CMakeLists.txt
+++ b/src/tbb/CMakeLists.txt
@@ -151,7 +151,7 @@ else()
     else()
         set(_includedir_for_pc_file "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
     endif()
-    configure_file(${CMAKE_SOURCE_DIR}/integration/pkg-config/tbb.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${TBB_LIB_NAME}.pc @ONLY)
+    configure_file(${PROJECT_SOURCE_DIR}/integration/pkg-config/tbb.pc.in ${CMAKE_CURRENT_BINARY_DIR}/${TBB_LIB_NAME}.pc @ONLY)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${TBB_LIB_NAME}.pc
             DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()

